METRICS.md

```diff
@@ -0,0 +1,352 @@
+                                      value=item[1]));
+        }
+        
+        metrics.append(new Metric(name='total', 
+                                  category=MetricCategory.NUM_REQUESTS, 
+                                  value=requestMetrics.total()));
+        return metrics;
+    }
+    
+    proc getResponseTimeMetrics() throws {
+        var metrics = new list(owned Metric?);
+
+        for item in responseTimeMetrics.items() {
+            metrics.append(new Metric(name=item[0], 
+                                      category=MetricCategory.RESPONSE_TIME,
+                                      value=item[1]));
+        }
+
+        return metrics;
+    }
+    
+    proc getServerMetrics() throws {
+        var metrics: list(owned Metric?);
+         
+        for item in serverMetrics.items(){
+            metrics.append(new Metric(name=item[0], category=MetricCategory.SERVER, 
+                                          value=item[1]));
+        }
+
+        return metrics;    
+    }    
+
+    proc getSystemMetrics() throws {
+        var metrics = new list(owned Metric?);
+
+        for loc in Locales {
+            var used = memoryUsed():real;
+            var total = loc.physicalMemory():real;
+            
+            mLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                              'memoryUsed: %i physicalMemory: %i'.format(used,total));
+
+            metrics.append(new LocaleMetric(name="arkouda_memory_used_per_locale",
+                             category=MetricCategory.SYSTEM,
+                             locale_num=loc.id,
+                             locale_name=loc.name,
+                             locale_hostname = loc.hostname,
+                             value=used):Metric);
+            metrics.append(new LocaleMetric(name="arkouda_percent_memory_used_per_locale",
+                             category=MetricCategory.SYSTEM,
+                             locale_num=loc.id,
+                             locale_name=loc.name,
+                             locale_hostname = loc.hostname,                             
+                             value=used/total * 100):Metric);                            
+        }
+        return metrics;
+    }
+```
+
+The MetricsMsg module is integrated into the Arkouda server side workflow within the MetricsServerDaemon class located in the [ServerDaemon](src/ServerDaemon) module. 
+
+## Enabling Metrics Capture and Export
+
+The arkouda_server startup command that enables metrics capture and export has two variants
+
+### Standard Arkouda Deployment
+
+In situations where Arkouda is not registered with an external system such as Kubernetes, the launch command is as follows. Note METRICS_SERVICE_PORT only has to be set if the port cannot be the default value of 5556.
+
+```
+export METRICS_SERVICE_PORT=6556
+
+./arkouda_server -nl 3 --ServerDaemon.daemonTypes=ServerDaemonType.DEFAULT,ServerDaemonType.METRICS
+```
```

Comment:
I'm assuming this will not work if there's a space in the comma separated list of arguments passed to `ServerDaemon.daemonTypes`, is this worth calling out?

Reply:
Yeah, I will add error checking and ensure docs detail this